### PR TITLE
TELCODOCS-1999 Documentation for VP OpenShift Framework

### PR DESCRIPTION
**Pull request by kquinn1204**
[TELCODOCS-1999]: Documentation for VP OpenShift Framework

Issue:https://issues.redhat.com/browse/TELCODOCS-1999

Link to docs preview:  http://451.docs-pr.validatedpatterns.io/learn/vp_openshift_framework/

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by mbaldessari** [comment]
This is an automated message:

You can preview this docs PR at http://451.docs-pr.validatedpatterns.io
Note that they get generated every five minutes, so please wait a bit.

**Comment by kquinn1204** [comment]
@mbaldessari @beekhof @abhatt-rh @danmacpherson - can you please review this PR that addresses https://issues.redhat.com/browse/TELCODOCS-1999. I am on then hunt for similar info for Ansible framework.

File: content/learn/vp_openshift_framework.adoc [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+=== Other combination examples include:
+
+. *`values-hub-Azure.yaml`* only apply this Azure tweak on the hub cluster.
+. *`values-4.16.yaml`* apply these OCP 4.16 tweaks to all cluster groups in this pattern.
+
+Current supported cloud providers include *`AWS`*, *`Azure`*, *`GCP`*, and *Nutanix*.
+
+== Applications & subscriptions
+
```
Comment (mbaldessari) [inline content/learn/vp_openshift_framework.adoc:262]: Should applications & subscriptions have some text here or should we change the indentation?

File: content/learn/vp_openshift_framework.adoc [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+== Environment values and Helm
+
+The reason the above values files exist is to take advantage of Helms ability to use templates and substitute values into your charts. This makes the pattern very portable.
```
Comment (mbaldessari) [inline content/learn/vp_openshift_framework.adoc:265]: Helm's?

**Review by mbaldessari — COMMENTED** [review]
lgtm to me overall. I have added a couple of thoughts/comments

File: content/learn/vp_openshift_framework.adoc (hunk 1 above)
Comment (kquinn1204) [reply content/learn/vp_openshift_framework.adoc:262]: Removed as seems out of place here

**Comment by mbaldessari** [comment]
Mmmh I suspect that htmltest is being a bit daft on the canonical link, prolly it does not realize that it should be pointed to the local build site so it looks it up but it is 404 because it is part of this change. I'll go ahead and merge this and verify if my theory was correct or not with another PR

**Comment by abhatt-rh** [comment]
> Mmmh I suspect that htmltest is being a bit daft on the canonical link, prolly it does not realize that it should be pointed to the local build site so it looks it up but it is 404 because it is part of this change. I'll go ahead and merge this and verify if my theory was correct or not with another PR

Yes, that's my theory too and it's what we have been doing (ignoring these) for rel-canonical link related errors thus far. They work fine after the PR is merged. But it might be better to fix it for good. Have a test PR in flight to test it. Thanks @mbaldessari

**Comment by leifmadsen** [comment]
> > Mmmh I suspect that htmltest is being a bit daft on the canonical link, prolly it does not realize that it should be pointed to the local build site so it looks it up but it is 404 because it is part of this change. I'll go ahead and merge this and verify if my theory was correct or not with another PR
> 
> Yes, that's my theory too and it's what we have been doing (ignoring these) for rel-canonical link related errors thus far. They work fine after the PR is merged. But it might be better to fix it for good. Have a test PR in flight to test it. Thanks @mbaldessari

I've never run into this kind of canonical link issue with asciidoc (although I've also not generated via Hugo previously). I wonder if using `xref` instead of `link` would actually work better here.

I did some local testing with `xref` and adding the appropriate `id` to the pages, and that seems to continue working with Hugo. When previewing locally, the links look slightly different with a better path. I might try submitting a change here to test things and see if that would resolve issues with invalid htmltest checks.

When writing product docs, we always used `xref` instead of `link` unless it was an external link out of the guide.

**Comment by leifmadsen** [comment]
Well I tried using xref instead of a link, and htmltest is just gonna htmltest I guess :)

**Pull request merged**